.github/workflows/test.yml

```diff
@@ -88,3 +97,69 @@ jobs:
         run: ant init
       - name: Test tests
         run: ant build-test
+
+  enumerated-tests:
+    timeout-minutes: 2
+    name: "Enumerate all tests from integration-tests folder to run them independently"
+    runs-on: ubuntu-24.04
+    outputs:
+      matrix: ${{ steps.enumerate.outputs.matrix }}
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Enumerate tests
+        id: enumerate
+        shell: bash
+        run: |
+          ITEMS=""
+
+          for file in integration-tests/*; do
+            ITEMS="\"$file\",$ITEMS"
+          done
+
+          ITEMS="${ITEMS%%,}"
+          ITEMS="[$ITEMS]"
+
+          echo "matxix=$ITEMS" >> $GITHUB_OUTPUT
+
+  integration-test:
+    needs: [build, enumerated-tests]
+    timeout-minutes: 15
+    name: "Integration Test ${{ matrix.java }} c-s ${{ matrix.test }}"
+    strategy:
+      fail-fast: false
+      matrix:
+        java: ["11", "17", "21"]
```

Comment:
I think we'll be o.k. doing those test only 21 that we are gonna use from now on

in general we should remove the old version, it's a waste of our time, if we move we moved.